src/asynqp/__init__.py

```diff
@@ -48,7 +48,13 @@ def connect(host='localhost',
         kwargs['port'] = port
 
     dispatcher = Dispatcher()
-    transport, protocol = yield from loop.create_connection(lambda: AMQP(dispatcher, loop), **kwargs)
+    try:
+        transport, protocol = yield from loop.create_connection(lambda: AMQP(dispatcher, loop), **kwargs)
+    except (ConnectionRefusedError, OSError):
+        # Throw a single exception instead of two
+        raise ConnectionRefusedError('Failed to connect - host: {} port: {}'
+                                     ' username: {} password: {} virtual_host: {}'
+                                     .format(host, port, username, password, virtual_host))
```

Comment:
Is this strictly necessary? I'd generally prefer the actual exception that was raised to get propagated out, as it makes it easier for client code to respond to the different errors. It also simplifies debugging.

In any case, I'd prefer this to use an explicit `raise from`:

```python
except (ConnectionRefusedError, OSError) as e:
        raise ConnectionRefusedError(...) from e
```